synapse/metrics/_exposition.py

```diff
@@ -74,7 +89,18 @@ def sample_line(line, name):
     )
 
 
-def generate_latest(registry):
+def nameify_sample(sample):
```

Comment:
for the record, if we have to actively do stuff to support old versions of deps, I'm quite happy for us to drop compatibility with those old versions, so in this case dropping support for promethus_cleint<0.4 would have been fine. 

I'm just asking that we don't bump our min dependencies as a matter-of-course.

Anyway, this looks good.